### PR TITLE
Update for Microsoft.StorageCache 2023-05-01

### PR DESCRIPTION
**Pull request by rebecca337**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by rebecca337** [comment]
@archerzz we have 2 failures with the checks, however, I don't see what our error is? I already followed the recommended steps with one failure and ran the Export-API script

File: sdk/storagecache/Azure.ResourceManager.StorageCache/src/Generated/AmlFilesystemCollection.cs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+using Azure.ResourceManager.Resources;
+
+namespace Azure.ResourceManager.StorageCache
+{
+    /// <summary>
+    /// A class representing a collection of <see cref="AmlFilesystemResource" /> and their operations.
+    /// Each <see cref="AmlFilesystemResource" /> in the collection will belong to the same instance of <see cref="ResourceGroupResource" />.
+    /// To get an <see cref="AmlFilesystemCollection" /> instance call the GetAmlFilesystems method from an instance of <see cref="ResourceGroupResource" />.
+    /// </summary>
+    public partial class AmlFilesystemCollection : ArmCollection, IEnumerable<AmlFilesystemResource>, IAsyncEnumerable<AmlFilesystemResource>
```
Comment (archerzz) [inline sdk/storagecache/Azure.ResourceManager.StorageCache/src/Generated/AmlFilesystemCollection.cs:27]: I think `Filesystem` should be `FileSystem`.
```suggestion
    public partial class AmlFileSystemCollection : ArmCollection, IEnumerable<AmlFileSystemResource>, IAsyncEnumerable<AmlFileSystemResource>
```

There are many places where `Filesystem` is used, which should be fixed. AFAICS, there are two options:
- fix the original swagger definitions
- or use batch renaming directive provided by us, see [here](https://github.com/Azure/azure-sdk-for-net/blob/c99853c16d6e167558e10dd56e2b98543c01617f/sdk/compute/Azure.ResourceManager.Compute/src/autorest.md?plain=1#L35-L49) for examples.

Comment (rebecca337) [reply sdk/storagecache/Azure.ResourceManager.StorageCache/src/Generated/AmlFilesystemCollection.cs:27]: @archerzz our swagger has already been merged with this naming convention: https://github.com/Azure/azure-rest-api-specs/pull/23563 . We had this reviewed and approved by ARM reviewers and our other generated SDKs follow this convention. I think "Filesystem" would be the better option?

Comment (archerzz) [reply sdk/storagecache/Azure.ResourceManager.StorageCache/src/Generated/AmlFilesystemCollection.cs:27]: @rebecca337 I think it's `File System`, so in [.Net SDK guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-properties), applying Pascal case will result in `FileSystem`.

Comment (rebecca337) [reply sdk/storagecache/Azure.ResourceManager.StorageCache/src/Generated/AmlFilesystemCollection.cs]: @archerzz I just fixed the naming for all AmlFilesystem references to AmlFileSystem

**Review by archerzz — APPROVED** [review]

**Pull request merged**